tephra-core/src/main/java/co/cask/tephra/distributed/CloseableThriftClient.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.tephra.distributed;
+
+import java.io.Closeable;
+import java.io.IOException;
+
+/**
+ * An {@link AutoCloseable} to automatically return the thrift client to the ThriftClientProvider.
+ */
+public class CloseableThriftClient implements Closeable {
```

Comment:
I think what I want to ask is, when `close()` is called, do you expect `IOException` or `Exception`? That's the major difference between those two.